From 7a3508ece9540fe81e2fb330e1eadf6293e1621f Mon Sep 17 00:00:00 2001
From: Johannes Thumshirn <jthumshirn@suse.de>
Date: Fri, 6 May 2016 14:58:57 +0200
Subject: [PATCH] PM / hibernate: Fix warning about non-void function not returning
Patch-mainline: Never, SUSE specific

hibernation_available() shall return bool, make it do so. Also remove the
potentially unwanted/dangerous second call to get_securelevel().

Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 kernel/power/hibernate.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/power/hibernate.c b/kernel/power/hibernate.c
index 19d8e38..f70f341 100644
--- a/kernel/power/hibernate.c
+++ b/kernel/power/hibernate.c
@@ -77,7 +77,7 @@ bool hibernation_available(void)
 
 	if (get_securelevel() <= 0)
 		return true;
-	else if (get_securelevel() > 0) {
+	else {
 #ifdef CONFIG_HIBERNATE_VERIFICATION
 		sigenforce = 1;
 		return true;
-- 
1.8.5.6

